### PR TITLE
Go gRPC, HTTP client and server interoperability tests implemented

### PR DESCRIPTION
**Pull request by odeke-em**

### CODE REVIEW

File: integration/go-grpc-client/interop_test.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+	// Since we cannot introspect tag.Map, we can only
+	// assert that its decoding returns the originally
+	// sent tag.Map, with the caveat: gRPC inserts the key "method" on sending
+	// to the serverside, so let's delete it from the serverside tagMap
+	inTagMap, err := tag.Decode(res.TagsBlob)
+	if err != nil {
+		t.Fatalf("Failed to decode tagBlob (% X) err: %v", res.TagsBlob, err)
+	}
+	tag.Delete(mustKey("method")).Mutate(inTagMap)
```
Comment (ramonza) [inline integration/go-grpc-client/interop_test.go]: this sounds like you found a bug. I don't think we should be sending a method tag - what is the point? would you mind opening an issue on opencensus-go?

File: integration/go-http-server/main.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	case "tracecontext":
+		fh = new(tracecontext.HTTPFormat)
+	}
+
+	atRuntimeHandler := &ochttp.Handler{
+		Handler:     mph.mux,
+		Propagation: fh,
+	}
+
+	atRuntimeHandler.ServeHTTP(w, r)
```
Comment (ramonza) [inline integration/go-http-server/main.go]: this is pretty weird - why not just set up separate routes that 1 for each propagation format?

File: integration/Makefile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+all: gen_proto
```
Comment (ramonza) [inline integration/Makefile]: can you add a `make test` to set everything up and run all the tests?

File: integration/README.md [hunk 4]
```diff
@@ -0,0 +1 @@
+## Interop testing
```
Comment (ramonza) [inline integration/README.md]: lets add instructions on running here, which ideally should just be "run make test"

File: integration/go-grpc-client/interop_test.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+	pb "github.com/census-instrumentation/opencensus-experiments/integration/proto"
+)
+
+var setups = []struct {
+	name         string
+	envAddrKey   string
+	fallbackAddr string
+}{
+	{name: "GoClient-GoServer", envAddrKey: "GO_SERVER_ADDR", fallbackAddr: ":9800"},
```
Comment (ramonza) [inline integration/go-grpc-client/interop_test.go]: kinda weird that this is being configured in Go: i would expect the server/client combination to be specified at the top level in a shell script / makefile so other languages wouldn't need to be concerned with editing Go files

**Comment by ramonza** [comment]
@odeke-em did you get contributor access to this repo yet?

File: integration/go-http-server/main.go (hunk 2 above)
Comment (odeke-em) [reply integration/go-http-server/main.go]: It was simpler and sufficient to automatically create these :)

File: integration/go-grpc-client/interop_test.go (hunk 1 above)
Comment (odeke-em) [reply integration/go-grpc-client/interop_test.go]: Sure, I'll file the bug in a second.

**Comment by odeke-em** [comment]
@ramonza unfortunately I haven't yet gotten contributor access, but I'll use my own fork in the meantime.

File: integration/go-grpc-client/interop_test.go (hunk 1 above)
Comment (odeke-em) [reply integration/go-grpc-client/interop_test.go]: I filed https://github.com/census-instrumentation/opencensus-go/issues/541

File: integration/README.md (hunk 4 above)
Comment (odeke-em) [reply integration/README.md]: Done, thank you!

**Comment by odeke-em** [comment]
Thank you for the review @ramonza! I'll merge this in to ensure that the Java team can bootstrap using the Go servers and the proto definitions as well as setup. We can keep iterating on this.

**Pull request merged**